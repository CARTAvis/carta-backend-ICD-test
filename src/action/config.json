{
    "serverURL": "wss://carta.asiaa.sinica.edu.tw/socketdev",
    "serverURL0": "ws://127.0.0.1:3002",
    "serverURL1": "ws://cartaviewer:5000",
    "path": {
        "root": ".",
        "base": "$BASE",
        "directory": ".",
        "QA": "set_QA",
        "region": "set_QA/set_QA_regionTest",
        "performance": "set_QA_performance"
    },
    "image": {
        "singleChannel": "cube_A/cube_A_01600_z00001.fits",
        "cube": "cube_A/cube_A_00800_z00100.fits"
    },
    "timeout": {
        "connection": 1000,
        "listFile": 1000,
        "openFile": 1000,
        "readFile": 6100,
        "readLargeImage": 60000,
        "changeChannel": 3000,
        "playImages": 8000,
        "cubeHistogram": 500000,
        "cancel": 5000,
        "concurrent": 4000,
        "messageEvent": 500,
        "mouseEvent": 400,
        "region": 500,
        "import": 500,
        "export": 500,
        "resume": 10000,
        "sleep": 3000,
        "playAnimator": 140000,
        "playLargeAnimator": 600000,
        "contour": 4000,
        "zoom": 12200
    },
    "repeat": {
        "concurrent": 10,
        "cursor": 20,
        "image": 10,
        "animation": 10,
        "resumeSession": 10,
        "contour": 10,
        "zoom": 5
    },
    "log": {
        "error": false,
        "warning": true,
        "event": false
    }
}