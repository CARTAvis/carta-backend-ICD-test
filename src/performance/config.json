{    
    "serverURL": "ws://127.0.0.1",
    "port": 7001,
    "timeout": {
        "connection": 2000,
        "openFile": 20000,
        "readFile": 40000
    },
    "wait": {
        "exec": 500,
        "ps": 1000,
        "reconnect": 30,
        "event": 35,
        "cursor": 35
    },
    "repeat": {
        "cursor": 20,
        "event": 10
    },
    "path": {
        "backend": "/Users/zarda/GitHub/carta-backend-nrao/build",
        "performanceTestResultDir": "/Users/zarda/GitHub/carta-backend-ICD-test/performance_result",
        "base": "$HOME/CARTA/Images",
        "performance": "set_QA_performance"
    },
    "log" : false
}