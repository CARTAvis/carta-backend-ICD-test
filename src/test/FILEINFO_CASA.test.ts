import { CARTA } from "carta-protobuf";
import { Client } from "./CLIENT";
import config from "./config.json";

let testServerUrl = config.serverURL;
let testSubdirectory = config.path.QA;
let connectTimeout = config.timeout.connection;
let listFileTimeout = config.timeout.listFile;
let openFileTimeout = config.timeout.openFile;

interface AssertItem {
    register: CARTA.IRegisterViewer;
    filelist: CARTA.IFileListRequest;
    fileInfoRequest: CARTA.IFileInfoRequest[];
    fileInfoResponse: CARTA.IFileInfoResponse[];
};

let assertItem: AssertItem = {
    register: {
        sessionId: 0,
        clientFeatureFlags: 5,
    },
    filelist: { directory: testSubdirectory },
    fileInfoRequest: [
        {
            directory: testSubdirectory,
            file: "M17_SWex.image",
            hdu: "",
        },
    ],
    fileInfoResponse: [
        {
            success: true,
            message: "",
            fileInfo: {
                name: "M17_SWex.image",
                type: CARTA.FileType.CASA,
                size: 53009869,
                HDUList: [""],
            },
            fileInfoExtended: {
                dimensions: 4,
                width: 640,
                height: 800,
                depth: 25,
                stokes: 1,
                stokesVals: [],
                computedEntries: [
                    { name: "Name", value: "M17_SWex.image" },
                    { name: "Shape", value: "[640, 800, 25, 1]" },
                    {
                        name: "Number of channels",
                        value: "25",
                        entryType: 2,
                        numericValue: 25
                    },
                    {
                        name: "Number of stokes",
                        value: "1",
                        entryType: 2,
                        numericValue: 1
                    },
                    { name: "Coordinate type", value: "Right Ascension, Declination" },
                    { name: "Projection", value: "SIN" },
                    { name: "Image reference pixels", value: "[321, 401]" },
                    { name: "Image reference coords", value: "[18:20:21.0000, -016.12.10.0000]" },
                    { name: "Image ref coords (deg)", value: "[275.088 deg, -16.2028 deg]" },
                    { name: "Celestial frame", value: "ICRS" },
                    { name: "Spectral frame", value: "LSRK" },
                    { name: "Velocity definition", value: "RADIO" },
                    { name: "Pixel unit", value: "Jy/beam" },
                    { name: "Pixel increment", value: "-0.4\", 0.4\"" },
                    { name: "Restoring beam", value: "2.06105\" X 1.49126\", -74.6267 deg" }
                ],
                headerEntries: [
                    { name: "SIMPLE", value: "T", entryType: 2, numericValue: 1, comment: "Standard FITS" },
                    { name: "BITPIX", value: "-32", entryType: 2, numericValue: -32, comment: "Floating point (32 bit)" },
                    { name: "NAXIS", value: "4", entryType: 2, numericValue: 4 },
                    { name: "NAXIS1", value: "640", entryType: 2, numericValue: 640 },
                    { name: "NAXIS2", value: "800", entryType: 2, numericValue: 800 },
                    { name: "NAXIS3", value: "25", entryType: 2, numericValue: 25 },
                    { name: "NAXIS4", value: "1", entryType: 2, numericValue: 1 },
                    {
                        name: "BSCALE",
                        value: "1.000000000000E+00",
                        entryType: 1,
                        numericValue: 1,
                        comment: "PHYSICAL = PIXEL*BSCALE + BZERO"
                    },
                    {
                        name: "BZERO",
                        value: "0.000000000000E+00",
                        entryType: 1,
                    },
                    {
                        name: "BMAJ",
                        value: "5.725136068132E-04",
                        entryType: 1,
                        numericValue: 0.0005725136068132
                    },
                    {
                        name: "BMIN",
                        value: "4.142385721207E-04",
                        entryType: 1,
                        numericValue: 0.0004142385721207
                    },
                    {
                        name: "BPA",
                        value: "-7.462673187256E+01",
                        entryType: 1,
                        numericValue: -74.62673187256
                    },
                    { name: "BTYPE", value: "Intensity" },
                    { name: "OBJECT", value: "M17SW" },
                    { name: "BUNIT", value: "Jy/beam", comment: "Brightness (pixel) unit" },
                    { name: "RADESYS", value: "ICRS" },
                    {
                        name: "LONPOLE",
                        value: "1.800000000000E+02",
                        entryType: 1,
                        numericValue: 180
                    },
                    {
                        name: "LATPOLE",
                        value: "-1.620277777779E+01",
                        entryType: 1,
                        numericValue: -16.20277777779
                    },
                    {
                        name: "PC1_1",
                        value: "1.000000000000E+00",
                        entryType: 1,
                        numericValue: 1
                    },
                    { name: "PC2_1", value: "0.000000000000E+00", entryType: 1 },
                    { name: "PC3_1", value: "0.000000000000E+00", entryType: 1 },
                    { name: "PC4_1", value: "0.000000000000E+00", entryType: 1 },
                    { name: "PC1_2", value: "0.000000000000E+00", entryType: 1 },
                    {
                        name: "PC2_2",
                        value: "1.000000000000E+00",
                        entryType: 1,
                        numericValue: 1
                    },
                    { name: "PC3_2", value: "0.000000000000E+00", entryType: 1 },
                    { name: "PC4_2", value: "0.000000000000E+00", entryType: 1 },
                    { name: "PC1_3", value: "0.000000000000E+00", entryType: 1 },
                    { name: "PC2_3", value: "0.000000000000E+00", entryType: 1 },
                    {
                        name: "PC3_3",
                        value: "1.000000000000E+00",
                        entryType: 1,
                        numericValue: 1
                    },
                    { name: "PC4_3", value: "0.000000000000E+00", entryType: 1 },
                    { name: "PC1_4", value: "0.000000000000E+00", entryType: 1 },
                    { name: "PC2_4", value: "0.000000000000E+00", entryType: 1 },
                    { name: "PC3_4", value: "0.000000000000E+00", entryType: 1 },
                    {
                        name: "PC4_4",
                        value: "1.000000000000E+00",
                        entryType: 1,
                        numericValue: 1
                    },
                    { name: "CTYPE1", value: "RA---SIN" },
                    {
                        name: "CRVAL1",
                        value: "2.750875000001E+02",
                        entryType: 1,
                        numericValue: 275.0875000001
                    },
                    {
                        name: "CDELT1",
                        value: "-1.111111111111E-04",
                        entryType: 1,
                        numericValue: -0.00011111111111110002
                    },
                    { name: "CRPIX1", value: "321", entryType: 1, numericValue: 321 },
                    { name: "CUNIT1", value: "deg" },
                    { name: "CTYPE2", value: "DEC--SIN" },
                    {
                        name: "CRVAL2",
                        value: "-1.620277777779E+01",
                        entryType: 1,
                        numericValue: -16.20277777779
                    },
                    {
                        name: "CDELT2",
                        value: "1.111111111111E-04",
                        entryType: 1,
                        numericValue: 0.00011111111111110002
                    },
                    { name: "CRPIX2", value: "401", entryType: 1, numericValue: 401 },
                    { name: "CUNIT2", value: "deg" },
                    { name: "CTYPE3", value: "FREQ" },
                    {
                        name: "CRVAL3",
                        value: "8.675139618840E+10",
                        entryType: 1,
                        numericValue: 86751396188.4
                    },
                    {
                        name: "CDELT3",
                        value: "-2.442377011414E+05",
                        entryType: 1,
                        numericValue: -244237.7011414
                    },
                    { name: "CRPIX3", value: "1", entryType: 1, numericValue: 1 },
                    { name: "CUNIT3", value: "Hz" },
                    { name: "CTYPE4", value: "STOKES" },
                    {
                        name: "CRVAL4",
                        value: "1.000000000000E+00",
                        entryType: 1,
                        numericValue: 1
                    },
                    {
                        name: "CDELT4",
                        value: "1.000000000000E+00",
                        entryType: 1,
                        numericValue: 1
                    },
                    { name: "CRPIX4", value: "1", entryType: 1, numericValue: 1 },
                    { name: "CUNIT4" },
                    { name: "PV2_1", value: "0.000000000000E+00", entryType: 1 },
                    { name: "PV2_2", value: "0.000000000000E+00", entryType: 1 },
                    {
                        name: "RESTFRQ",
                        value: "8.675429000000E+10",
                        entryType: 1,
                        numericValue: 86754290000,
                        comment: "Rest Frequency (Hz)"
                    },
                    { name: "SPECSYS", value: "LSRK", comment: 'Spectral reference frame' },
                    {
                        name: "ALTRVAL",
                        value: "9.999999914171E+03",
                        entryType: 1,
                        numericValue: 9999.99991417096
                    },
                    { name: "ALTRPIX", value: "1", entryType: 1, numericValue: 1 },
                    {
                        name: "VELREF",
                        value: "257",
                        entryType: 2,
                        numericValue: 257,
                        comment: '1 LSR, 2 HEL, 3 OBS, +256 Radio'
                    },
                    { name: "TELESCOP", value: "ALMA" },
                    { name: "OBSERVER", value: "sishii" },
                    { name: "DATE-OBS", value: "2016-04-03T13:02:58.800000" },
                    { name: "TIMESYS", value: "UTC" },
                    {
                        name: "OBSRA",
                        value: "2.750875000001E+02",
                        entryType: 1,
                        numericValue: 275.0875000001
                    },
                    {
                        name: "OBSDEC",
                        value: "-1.620277777779E+01",
                        entryType: 1,
                        numericValue: -16.202777777790004
                    },
                    {
                        name: "OBSGEO-X",
                        value: "2.225142180269E+06",
                        entryType: 1,
                        numericValue: 2225142.1802689997
                    },
                    {
                        name: "OBSGEO-Y",
                        value: "-5.440307370349E+06",
                        entryType: 1,
                        numericValue: -5440307.370349
                    },
                    {
                        name: "OBSGEO-Z",
                        value: "-2.481029851874E+06",
                        entryType: 1,
                        numericValue: -2481029.851874
                    },
                ],
            },
        },
    ],
};

describe("FILEINFO test: Testing if info of an image file is correctly delivered by the backend", () => {

    let Connection: Client;
    beforeAll(async () => {
        Connection = new Client(testServerUrl);
        await Connection.open();
        await Connection.send(CARTA.RegisterViewer, assertItem.register);
        await Connection.receive(CARTA.RegisterViewerAck);
    }, connectTimeout);

    describe(`Go to "${testSubdirectory}" folder`, () => {
        beforeAll(async () => {
            await Connection.send(CARTA.FileListRequest, assertItem.filelist);
            await Connection.receive(CARTA.FileListResponse);
        }, listFileTimeout);

        describe(`query the info of file : ${assertItem.fileInfoRequest[0].file}`, () => {
            let FileInfoResponseTemp: CARTA.FileInfoResponse;
            test(`FILE_INFO_RESPONSE should arrive within ${openFileTimeout} ms".`, async () => {
                await Connection.send(CARTA.FileInfoRequest, assertItem.fileInfoRequest[0]);
                FileInfoResponseTemp = await Connection.receive(CARTA.FileInfoResponse);
            }, openFileTimeout);

            test("FILE_INFO_RESPONSE.success = true", () => {
                expect(FileInfoResponseTemp.success).toBe(true);
            });

            test(`FILE_INFO_RESPONSE.file_info.HDU_List = [${assertItem.fileInfoResponse[0].fileInfo.HDUList}]`, () => {
                expect(FileInfoResponseTemp.fileInfo.HDUList[0]).toBe(assertItem.fileInfoResponse[0].fileInfo.HDUList[0]);
            });

            test(`FILE_INFO_RESPONSE.file_info.name = "${assertItem.fileInfoResponse[0].fileInfo.name}"`, () => {
                expect(FileInfoResponseTemp.fileInfo.name).toEqual(assertItem.fileInfoResponse[0].fileInfo.name);
            });

            test(`FILE_INFO_RESPONSE.file_info.size = ${assertItem.fileInfoResponse[0].fileInfo.size}`, () => {
                expect(FileInfoResponseTemp.fileInfo.size.toString()).toEqual(assertItem.fileInfoResponse[0].fileInfo.size.toString());
            });

            test(`FILE_INFO_RESPONSE.file_info.type = ${CARTA.FileType.CASA}`, () => {
                expect(FileInfoResponseTemp.fileInfo.type).toBe(assertItem.fileInfoResponse[0].fileInfo.type);
            });

            test(`FILE_INFO_RESPONSE.file_info_extended.dimensions = ${assertItem.fileInfoResponse[0].fileInfoExtended.dimensions}`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.dimensions).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.dimensions);
            });

            test(`FILE_INFO_RESPONSE.file_info_extended.width = ${assertItem.fileInfoResponse[0].fileInfoExtended.width}`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.width).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.width);
            });

            test(`FILE_INFO_RESPONSE.file_info_extended.height = ${assertItem.fileInfoResponse[0].fileInfoExtended.height}`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.height).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.height);
            });

            if (assertItem.fileInfoResponse[0].fileInfoExtended.dimensions > 2) {
                test(`FILE_INFO_RESPONSE.file_info_extended.depth = ${assertItem.fileInfoResponse[0].fileInfoExtended.depth}`, () => {
                    expect(FileInfoResponseTemp.fileInfoExtended.depth).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.depth);
                });
            };

            if (assertItem.fileInfoResponse[0].fileInfoExtended.dimensions > 3) {
                test(`FILE_INFO_RESPONSE.file_info_extended.stokes = ${assertItem.fileInfoResponse[0].fileInfoExtended.stokes}`, () => {
                    expect(FileInfoResponseTemp.fileInfoExtended.stokes).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.stokes);
                });
            };

            test(`FILE_INFO_RESPONSE.file_info_extended.stokes_vals = [${assertItem.fileInfoResponse[0].fileInfoExtended.stokesVals}]`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.stokesVals).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.stokesVals);
            });

            test(`len(FILE_INFO_RESPONSE.file_info_extended.computed_entries)==${assertItem.fileInfoResponse[0].fileInfoExtended.computedEntries.length}`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.computedEntries.length).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.computedEntries.length);
            });

            test(`assert FILE_INFO_RESPONSE.file_info_extended.computed_entries`, () => {
                assertItem.fileInfoResponse[0].fileInfoExtended.computedEntries.map((entry: CARTA.IHeaderEntry, index) => {
                    expect(FileInfoResponseTemp.fileInfoExtended.computedEntries).toContainEqual(entry);
                });
            });

            test(`len(file_info_extended.header_entries)==${assertItem.fileInfoResponse[0].fileInfoExtended.headerEntries.length}`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.headerEntries.length).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.headerEntries.length)
            });

            test(`assert FILE_INFO_RESPONSE.file_info_extended.header_entries`, () => {
                assertItem.fileInfoResponse[0].fileInfoExtended.headerEntries.map((entry: CARTA.IHeaderEntry, index) => {
                    console.log(FileInfoResponseTemp.fileInfoExtended.headerEntries)
                    expect(FileInfoResponseTemp.fileInfoExtended.headerEntries).toContainEqual(entry);
                });
            });
        });
    });
    afterAll(() => Connection.close());
});