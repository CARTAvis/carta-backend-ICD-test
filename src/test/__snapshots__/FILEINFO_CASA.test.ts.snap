// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FILEINFO_CASA: Testing if info of an CASA image file is correctly delivered by the backend Go to "set_QA" folder query the info of file : M17_SWex.image assert FILE_INFO_RESPONSE.file_info_extended.computed_entries 1`] = `
Array [
  Object {
    "name": "Name",
    "value": "M17_SWex.image",
  },
  Object {
    "name": "Shape",
    "value": "[640, 800, 25, 1]",
  },
  Object {
    "entryType": "INT",
    "name": "Number of channels",
    "numericValue": 25,
    "value": "25",
  },
  Object {
    "entryType": "INT",
    "name": "Number of polarizations",
    "numericValue": 1,
    "value": "1",
  },
  Object {
    "name": "Coordinate type",
    "value": "Right Ascension, Declination",
  },
  Object {
    "name": "Projection",
    "value": "SIN",
  },
  Object {
    "name": "Image reference pixels",
    "value": "[321, 401]",
  },
  Object {
    "name": "Image reference coords",
    "value": "[18:20:21.0000, -016.12.10.0000]",
  },
  Object {
    "name": "Image ref coords (deg)",
    "value": "[275.088 deg, -16.2028 deg]",
  },
  Object {
    "name": "Pixel increment",
    "value": "-0.4\\", 0.4\\"",
  },
  Object {
    "name": "Pixel unit",
    "value": "Jy/beam",
  },
  Object {
    "name": "Celestial frame",
    "value": "ICRS",
  },
  Object {
    "name": "Spectral frame",
    "value": "LSRK",
  },
  Object {
    "name": "Velocity definition",
    "value": "RADIO",
  },
  Object {
    "name": "Restoring beam",
    "value": "2.06105\\" X 1.49126\\", -74.6267 deg",
  },
]
`;

exports[`FILEINFO_CASA: Testing if info of an CASA image file is correctly delivered by the backend Go to "set_QA" folder query the info of file : M17_SWex.image assert FILE_INFO_RESPONSE.file_info_extended.header_entries 1`] = `
Array [
  Object {
    "comment": "IMAGE extension",
    "name": "XTENSION",
    "value": "IMAGE",
  },
  Object {
    "comment": "Floating point (32 bit)",
    "entryType": "INT",
    "name": "BITPIX",
    "numericValue": -32,
    "value": "-32",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS",
    "numericValue": 4,
    "value": "4",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS1",
    "numericValue": 640,
    "value": "640",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS2",
    "numericValue": 800,
    "value": "800",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS3",
    "numericValue": 25,
    "value": "25",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS4",
    "numericValue": 1,
    "value": "1",
  },
  Object {
    "entryType": "INT",
    "name": "PCOUNT",
    "value": "0",
  },
  Object {
    "entryType": "INT",
    "name": "GCOUNT",
    "numericValue": 1,
    "value": "1",
  },
  Object {
    "comment": "PHYSICAL = PIXEL*BSCALE + BZERO",
    "entryType": "FLOAT",
    "name": "BSCALE",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "BZERO",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "BMAJ",
    "numericValue": 0.0005725136068132,
    "value": "5.725136068132E-04",
  },
  Object {
    "entryType": "FLOAT",
    "name": "BMIN",
    "numericValue": 0.0004142385721207,
    "value": "4.142385721207E-04",
  },
  Object {
    "entryType": "FLOAT",
    "name": "BPA",
    "numericValue": -74.62673187256,
    "value": "-7.462673187256E+01",
  },
  Object {
    "name": "BTYPE",
    "value": "Intensity",
  },
  Object {
    "name": "OBJECT",
    "value": "M17SW",
  },
  Object {
    "comment": "Brightness (pixel) unit",
    "name": "BUNIT",
    "value": "Jy/beam",
  },
  Object {
    "name": "RADESYS",
    "value": "ICRS",
  },
  Object {
    "entryType": "FLOAT",
    "name": "LONPOLE",
    "numericValue": 180,
    "value": "1.800000000000E+02",
  },
  Object {
    "entryType": "FLOAT",
    "name": "LATPOLE",
    "numericValue": -16.20277777779,
    "value": "-1.620277777779E+01",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC1_1",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC2_1",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC3_1",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC4_1",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC1_2",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC2_2",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC3_2",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC4_2",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC1_3",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC2_3",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC3_3",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC4_3",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC1_4",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC2_4",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC3_4",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC4_4",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "name": "CTYPE1",
    "value": "RA---SIN",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRVAL1",
    "numericValue": 275.0875000001,
    "value": "2.750875000001E+02",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CDELT1",
    "numericValue": -0.00011111111111110002,
    "value": "-1.111111111111E-04",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRPIX1",
    "numericValue": 321,
    "value": "321",
  },
  Object {
    "name": "CUNIT1",
    "value": "deg",
  },
  Object {
    "name": "CTYPE2",
    "value": "DEC--SIN",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRVAL2",
    "numericValue": -16.20277777779,
    "value": "-1.620277777779E+01",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CDELT2",
    "numericValue": 0.00011111111111110002,
    "value": "1.111111111111E-04",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRPIX2",
    "numericValue": 401,
    "value": "401",
  },
  Object {
    "name": "CUNIT2",
    "value": "deg",
  },
  Object {
    "name": "CTYPE3",
    "value": "FREQ",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRVAL3",
    "numericValue": 86751396188.4,
    "value": "8.675139618840E+10",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CDELT3",
    "numericValue": -244237.7011414,
    "value": "-2.442377011414E+05",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRPIX3",
    "numericValue": 1,
    "value": "1",
  },
  Object {
    "name": "CUNIT3",
    "value": "Hz",
  },
  Object {
    "name": "CTYPE4",
    "value": "STOKES",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRVAL4",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CDELT4",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRPIX4",
    "numericValue": 1,
    "value": "1",
  },
  Object {
    "name": "CUNIT4",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PV2_1",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PV2_2",
    "value": "0.000000000000E+00",
  },
  Object {
    "comment": "Rest Frequency (Hz)",
    "entryType": "FLOAT",
    "name": "RESTFRQ",
    "numericValue": 86754290000,
    "value": "8.675429000000E+10",
  },
  Object {
    "comment": "Spectral reference frame",
    "name": "SPECSYS",
    "value": "LSRK",
  },
  Object {
    "entryType": "FLOAT",
    "name": "ALTRVAL",
    "numericValue": 9999.99991417096,
    "value": "9.999999914171E+03",
  },
  Object {
    "entryType": "FLOAT",
    "name": "ALTRPIX",
    "numericValue": 1,
    "value": "1",
  },
  Object {
    "comment": "1 LSR, 2 HEL, 3 OBS, +256 Radio",
    "entryType": "INT",
    "name": "VELREF",
    "numericValue": 257,
    "value": "257",
  },
  Object {
    "name": "TELESCOP",
    "value": "ALMA",
  },
  Object {
    "name": "OBSERVER",
    "value": "sishii",
  },
  Object {
    "name": "DATE-OBS",
    "value": "2016-04-03T13:02:58.800000",
  },
  Object {
    "name": "TIMESYS",
    "value": "UTC",
  },
  Object {
    "entryType": "FLOAT",
    "name": "OBSRA",
    "numericValue": 275.0875000001,
    "value": "2.750875000001E+02",
  },
  Object {
    "entryType": "FLOAT",
    "name": "OBSDEC",
    "numericValue": -16.202777777790004,
    "value": "-1.620277777779E+01",
  },
  Object {
    "entryType": "FLOAT",
    "name": "OBSGEO-X",
    "numericValue": 2225142.1802689997,
    "value": "2.225142180269E+06",
  },
  Object {
    "entryType": "FLOAT",
    "name": "OBSGEO-Y",
    "numericValue": -5440307.370349,
    "value": "-5.440307370349E+06",
  },
  Object {
    "entryType": "FLOAT",
    "name": "OBSGEO-Z",
    "numericValue": -2481029.851874,
    "value": "-2.481029851874E+06",
  },
]
`;
